src/serac/physics/utilities/finite_element_state.hpp

```diff
@@ -196,6 +196,14 @@ class FiniteElementState {
     return std::make_unique<Tensor>(&retrieve(space_));
   }
 
+  /**
+   * @brief Calculate the Lp norm of the underlying state grid function
+   *
+   * @param p Order of the norm
+   * @return The norm value
+   */
+  double norm(double p = 2) const;
```

Comment:
When practical, I prefer free functions and operator overloads over member functions for basic mathematical operations, since it is closer to the original mathematical notation (although mfem has mostly uses member functions). For example,
```cpp
double x_norm = norm(solid_solver.displacement());
```
is closer to ||u|| than 
```cpp
double x_norm = solid_solver.displacement().norm();
```
Making everything into member functions is more of a Java convention, and tends to hurt readability (in my opinion):
```java
double result = a.multiply(x.pow(2)).plus(b.multiply(x)).plus(c);
```
vs.
```cpp
double result = a * x * x + b * x + c;
```

This mostly just aesthetics, though. Does anyone else have a preference here?

Reply:
I agree, this is nicer. I'll go fix it up.